code/core/src/main/java/com/adobe/marketing/mobile/internal/eventhub/history/AndroidEventHistory.kt

```diff
@@ -131,29 +114,29 @@ internal class AndroidEventHistory : EventHistory {
      * match the contents of the [EventHistoryRequest] array.
      *
      * @param eventHistoryRequests an array of `EventHistoryRequest`s to be deleted
-     * @param handler a callback which will be called with a `int` containing the total number
+     * @param callback which will be called with a `int` containing the total number
      * of rows deleted from the `AndroidEventHistoryDatabase`
      */
     override fun deleteEvents(
         eventHistoryRequests: Array<out EventHistoryRequest>,
-        handler: EventHistoryResultHandler<Int>?
+        callback: AdobeCallbackWithError<Int>?
```

Comment:
When database errors occur, the `AndroidEventHistoryDatabase.delete` function suppresses the exception and returns a `0`. Should we standardize it with the other get and record functions that return a `-1` to signify a database error?

Reply:
Good catch I will change it to return -1

Reply:
I checked with iOS implementation of this. Looks like even there `0` is returned in case of database error which means no records were deleted from event history. We might want to either keep this as is or change it on both platforms. Thoughts?
https://github.com/adobe/aepsdk-core-ios/blob/main/AEPCore/Sources/eventhub/history/EventHistoryDatabase.swift#L148-L151

Reply:
I see. Given that we don't have a plan to retry today when the database errors occur, I think it should be fine to leave it as it is. Can you update the function docs to reflect this expected behavior?
@prudrabhat  Let us know if you have other thoughts.

Reply:
I changed this to call `fail` if an error occurs